core/HoistModel.js

```diff
@@ -4,39 +4,25 @@
  *
  * Copyright © 2020 Extremely Heavy Industries Inc.
  */
-import {ManagedSupport, ReactiveSupport, XhIdSupport, PersistSupport} from '@xh/hoist/core';
-import {applyMixin} from '@xh/hoist/utils/js';
-import {isFunction, isString} from 'lodash';
+import {HoistBase} from './HoistBase';
 
 /**
- * Core decorator for State Models in Hoist.
+ * Core class for State Models in Hoist.
  *
- * All State models in Hoist applications should typically be decorated with this function.
  * Adds support for managed events and mobx reactivity.
  *
  * A common use of HoistModel is to serve as a backing store for a HoistComponent.  Furthermore, if
  * a model is *created* by a HoistComponent it is considered to be 'owned' (or "hosted") by that
  * component.  An owned model will be automatically destroyed when its component is unmounted.
  *
- * For HoistModels that need to load/refresh data consider implementing LoadSupport.
- * This decorator will load data into the model when its component is first mounted, and will
- * register the model with the nearest RefreshContextModel for subsequent refreshes.
- *
- * @see LoadSupport
+ * HoistModels that need to load/refresh may implement doLoadAsync().  See HoistBase for more
+ * information.  If specified, this method will be used to load data into the model when its
+ * component is first mounted.  It will also register the model with the nearest
+ * RefreshContextModel for subsequent refreshes.
  */
-export function HoistModel(C) {
-    return applyMixin(C, {
-        name: 'HoistModel',
-        includes: [ManagedSupport, ReactiveSupport, PersistSupport, XhIdSupport],
-
-        provides: {
-            matchesSelector(selector) {
-                if (isFunction(selector)) return selector.isHoistModel ? this instanceof selector : selector(this);
-                if (isString(selector)) return !!this['is' + selector];
-                return false;
-            }
-        }
-    });
-}
+export class HoistModel extends HoistBase {
 
+    get isHoistModel() {return true}
 
+}
+HoistModel.isHoistModel = true;
```

Comment:
~~This is interesting - what's the purpose of installing this static on the class?~~

For use in the model context selector below, is that right? Worth a comment?

Reply:
I see we have this pattern consistently in place - don't think it needs a comment so much now that I understand it better and have read further down.

Any concerns having a static class property and the instance getter of the same name?

Reply:
no concerns with having the two flags be the same -- its a feature -- if you don't really care (or understand) the difference between whether you have an instance or the class, you will still get the right question to the answer you are asking.

It's what we did with the marker flags that were supplied by the mixin handling.